main/acle.md

```diff
@@ -4932,6 +5044,1381 @@ assertions, which may speed up access to aligned data (and will fault
+    ptr2 = f1;    // OK
+    ptr2 = f2;    // OK
+  }
+```
+
+## SME functions
+
+The functions in this section are defined by the header file
+[`<arm_sme.h>`](#arm_sme.h).  Except where noted otherwise,
+the functions do not have linkage.
+
+### SME PSTATE functions
+
+#### Prototypes
+
+``` c
+  __attribute__((arm_streaming_compatible))
+  bool __arm_in_streaming_mode(void);
+
+  // Has linkage.
+  __attribute__((arm_streaming_compatible))
+  void __arm_za_disable(void);
+```
+
+#### Semantics
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the section
+following it. --><span id="__arm_in_streaming_mode"></span>
+
+**`__arm_in_streaming_mode()`**
+
+> This call returns true if the [abstract machine](#abstract-machine) is
+> in streaming mode.  It always returns true when called from [streaming
+> statements](#streaming-statement) and it always return false when
+> called from [non-streaming statements](#non-streaming-statement).
+> However, the call is not semantically a constant expression even in
+> those cases.
+
+<!-- Do not remove the following `span`, they are needed to create an
+anchor that can be referred via an internal hyperlink to the section
+following it. --><span id="__arm_za_disable"></span>
+
+**`__arm_za_disable()`**
+
+> This call commits any pending lazy save and turns ZA off;
+> see [[AAPCS64]](#AAPCS64) for details.
+>
+> Note that since this is a [private-ZA](#private-za) function,
+> it is valid (though pointless) to call it from a function that has
+> [ZA state](#za-state). The compiler would simply reenable ZA after the
+> call and reload the saved ZA state.
+
+### SME ZA state assertions
+
+#### Prototypes
+
+``` c
+  __attribute__((arm_streaming_compatible, arm_shared_za))
+  void svundef_za();
+```
+
+#### Semantics
+
+**`svundef_za()`**
+
+> This call asserts that ZA does not currently have any useful data.
+> Semantically, it fills ZA with unpredictable data, although from
+> a quality of implementation perspective, it should not generate
+> any object code.
+>
+> The call can be used as an optimization hint to the compiler, so that
+> the compiler does not insert unnecessary code to save and restore the
+> current ZA contents. The call might also be useful for static analysis.
+
+### SME intrinsic functions
+
+#### Common rules
+
+The functions in this section have the following properties in common:
+
+*   Every argument named `tile`, `slice_offset` or `tile_mask` must
+    be an integer constant expression.
+
+*   ZA loads and stores do not use typed pointers, since there is
+    no C or C++ type information associated with the contents of ZA.
+
+*   Intrinsic functions have a `_hor` suffix if they operate on
+    horizontal slices of a given ZA tile and a `_ver` suffix if they
+    operate on vertical slices of a given ZA tile.
+
+#### LD1B, LD1H, LD1W, LD1D, LD1Q
+
+``` c
+  // Also for _za16, _za32, _za64 and _za128.
```

Comment:
Do they have the same signature of `svld1_hor_za8`? If so, worth specifying this fact in the comment?

Reply:
Yeah, I've added a note to say that these prototypes are unaffected by the type suffix.  I've also described in the MOVA section how those prototypes vary with the type suffix.